health.go

```diff
@@ -312,7 +312,15 @@ func (h *Health) safeUpdateState(stateEntry *State) {
 func (h *Health) safeGetStates() map[string]State {
 	h.statesLock.Lock()
 	defer h.statesLock.Unlock()
-	return h.states
+
+	// deep copy h.states to avoid race
+	statesCopy := make(map[string]State, 0)
+
+	for k, v := range h.states {
+		statesCopy[k] = v
+	}
+
+	return statesCopy
```

Comment:
I'm ok with this.  Copying a state doesnt produce any side-effects.
I would probably move the defer unlock to an explicit unlock right after the loop.  That's a micro-optimization, so it's up to you.